### PR TITLE
Use JSON Logging

### PR DESCRIPTION
**Pull request by vthacker**
Makes looking at logs in kibana much easier.

### CODE REVIEW

File: kaldb/pom.xml [hunk 1]
```diff
@@ -487,7 +492,14 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>2.4.1</version>
+                <version>3.2.4</version>
+                <dependencies>
+                    <dependency>
```
Comment (mansu) [inline kaldb/pom.xml:497]: Do we need these external dependencies? Can we make this change without these additional dependencies?

Comment (mansu) [reply kaldb/pom.xml:497]: How hard would it be to emit log messages in span json format?

File: kaldb/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -4,7 +4,9 @@
 
     <Appenders>
         <Console name="console" target="SYSTEM_OUT">
-            <PatternLayout  pattern="[%-5level] %d{yyyy-MM-dd HH:mm:ss.SSS} [%t] %notEmpty{[trace=%X{traceId}/%X{spanId}]} %c{1} - %msg%n" />
+            <!-- https://github.com/apache/logging-log4j2/blob/master/log4j-layout-template-json/src/main/resources/EcsLayout.json -->
+            <!-- MDC keys will get their own field as per EcsLayout.json  -->
+            <JsonTemplateLayout eventTemplateUri="classpath:EcsLayout.json" />
```
Comment (mansu) [inline kaldb/src/main/resources/log4j2.xml:9]: How hard would it be to emit log messages in span json format and use that as a template?

File: kaldb/pom.xml (hunk 1 above)
Comment (vthacker) [reply kaldb/pom.xml:497]: nope :) It won't compile. I've linked the github issue

File: kaldb/src/main/resources/log4j2.xml (hunk 2 above)
Comment (vthacker) [reply kaldb/src/main/resources/log4j2.xml:9]: What's a span JSON format? Also we have to then see if that format can be ingested by our internal logging pipeline.

File: kaldb/src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -4,7 +4,8 @@
 
     <Appenders>
         <Console name="console" target="SYSTEM_OUT">
-            <PatternLayout  pattern="[%-5level] %d{yyyy-MM-dd HH:mm:ss.SSS} [%t] %notEmpty{[trace=%X{traceId}/%X{spanId}]} %c{1} - %msg%n" />
+            <!-- https://github.com/apache/logging-log4j2/blob/master/log4j-layout-template-json/src/main/resources/EcsLayout.json -->
```
Comment (bryanlb) [inline kaldb/src/main/resources/log4j2.xml:7]: Should this comment be updated/removed?

**Review by bryanlb — APPROVED** [review]

File: kaldb/src/main/resources/log4j2.xml (hunk 3 above)
Comment (vthacker) [reply kaldb/src/main/resources/log4j2.xml:7]: I kinda removed it then kept it.. It helped me checking out examples from the project.

**Pull request merged**